### ISSUE TITLE
Locks all databases

### ISSUE THREAD

**Issue by anotherdevs** [hankinsoft/SQLPro#106]
Hi

It looks like MSSQL Pro locks all databases, just by having an active connection.
This prevents renaming other databases.

Is that an issue that can be looked into?

SQL to reproduce trace of open processes:

```
SELECT DB_NAME(dbid)AS ConnectedToDB,
hostname, program_name,loginame,
cpu, physical_io, memusage, login_time,
last_batch, [status]
FROM master.dbo.sysprocesses
ORDER BY dbid, login_time, last_batch
```

**Comment by anotherdevs** [comment]
SELECT DB_NAME(dbid)AS ConnectedToDB,
hostname, program_name,loginame,
cpu, physical_io, memusage, login_time,
last_batch, [status]
FROM master.dbo.sysprocesses
ORDER BY dbid, login_time, last_batch

**Comment by anotherdevs** [comment]
@hankinsoft Is that something you can look into?

**Comment by hankinsoft** [comment]
Hi,
> It looks like MSSQL Pro locks all databases, just by having an active connection.
This prevents renaming other databases.

What is the exact error you are seeing?

You posted a query for showing the current running queries.... Are you seeing locked queries when running this? If so could you please provide the output of the query?

Are you specifying a database name in your connection? If so is this the database you are attempting to rename?

Thanks!

**Comment by anotherdevs** [comment]
Hi @hankinsoft 

When opening a connection to a server it also makes a connection to every database. If someone wants to rename or drop a specific database they can't because the SQLPro for MSSQL has a open process towards the database.

You can try to run the SQL query on your server and see that it opens connections towards all databases.

You need to use the approach Navicat and SQL Server Management uses where you will need to right click (or double click) to create an active progress towards the database.

**Comment by hankinsoft** [comment]
Thanks. SQLPro will connect to each database in order to load details for filtering. It looks like at the moment it keeps the connections open, but I'll take a look into closing the connections after it loads the filtering deals. I believe that should sort the issue you are seeing.

**Comment by anotherdevs** [comment]
Hi @hankinsoft 

It certainly will. Can you ping me when you have published a new version?

**Comment by hankinsoft** [comment]
I’m hoping to have a build available later today. Are you using SQLPro Studio or SQLPro for MSSQL?

**Comment by anotherdevs** [comment]
Hi @hankinsoft 

I am using SQLPro for MSSQL.

**Comment by anotherdevs** [comment]
Hi @hankinsoft 

It's looking good. Seems to work perfectly.

Only problem now is that after you opened a database it still keeps an active process towards the database, even after you have "untoggled" the database.

However it's still a massive improvement.

**Comment by anotherdevs** [comment]
Hi again @hankinsoft 

I can't get filterering to work. It do not cache the databases, it seems like.

**Comment by hankinsoft** [comment]
Hi. Filtering has been addressed in item #48.
